crates/nu-command/src/conversions/into/duration.rs

```diff
@@ -0,0 +1,257 @@
+                                span,
+                            }],
+                            span,
+                        },
+                        Value::Record {
+                            cols: vec!["value".to_string()],
+                            vals: vec![Value::Duration {
+                                val: 2 * 60 * 1000 * 1000 * 1000,
+                                span,
+                            }],
+                            span,
+                        },
+                        Value::Record {
+                            cols: vec!["value".to_string()],
+                            vals: vec![Value::Duration {
+                                val: 3 * 60 * 60 * 1000 * 1000 * 1000,
+                                span,
+                            }],
+                            span,
+                        },
+                        Value::Record {
+                            cols: vec!["value".to_string()],
+                            vals: vec![Value::Duration {
+                                val: 4 * 24 * 60 * 60 * 1000 * 1000 * 1000,
+                                span,
+                            }],
+                            span,
+                        },
+                        Value::Record {
+                            cols: vec!["value".to_string()],
+                            vals: vec![Value::Duration {
+                                val: 5 * 7 * 24 * 60 * 60 * 1000 * 1000 * 1000,
+                                span,
+                            }],
+                            span,
+                        },
+                    ],
+                    span,
+                }),
+            },
+            Example {
+                description: "Convert string to duration",
+                example: "'7min' | into duration",
+                result: Some(Value::Duration {
+                    val: 7 * 60 * 1000 * 1000 * 1000,
+                    span,
+                }),
+            },
+        ]
+    }
+}
+
+fn into_duration(
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    let head = call.head;
+    let column_paths: Vec<CellPath> = call.rest(engine_state, stack, 0)?;
+
+    input.map(
+        move |v| {
+            if column_paths.is_empty() {
+                action(&v, head)
+            } else {
+                let mut ret = v;
+                for path in &column_paths {
+                    let r =
+                        ret.update_cell_path(&path.members, Box::new(move |old| action(old, head)));
+                    if let Err(error) = r {
+                        return Value::Error { error };
+                    }
+                }
+
+                ret
+            }
+        },
+        engine_state.ctrlc.clone(),
+    )
+}
+
+fn string_to_duration(s: &str, span: Span) -> Result<i64, ShellError> {
```

Comment:
There's a `parse_duration` in the `nu-parser` sub-crate you could have do this work for you. Right now it does multiple steps to get the parse, but you could split it into two functions. The second one would be called after the `let bytes = working_set.get_span_contents(span);` step and would pass it the bytes to parse.

Then you could export this function and use it here to parse the duration for you. From here, you would take the string, turn it to bytes, and pass it into the newly-exported function you created in nu-parser.

Reusing the parsing logic for both the main language and for `into duration` helps us in case we ever add support for other kinds of duration.

Reply:
Thank you for the details! I made a change.